diff -r dc4b49011838 dmenu.c
--- a/dmenu.c	Fri Apr 06 16:38:01 2012 +0100
+++ b/dmenu.c	Wed Apr 18 17:08:32 2012 +0200
@@ -271,7 +271,7 @@
 		case XK_w: /* delete word */
 			while(cursor > 0 && text[nextrune(-1)] == ' ')
 				insert(NULL, nextrune(-1) - cursor);
-			while(cursor > 0 && text[nextrune(-1)] != ' ')
+			while(cursor > 0 && text[nextrune(-1)] != ' ' && text[nextrune(-1)] != '/')
 				insert(NULL, nextrune(-1) - cursor);
 			break;
 		case XK_y: /* paste selection */
