diff -r f42a652fcdc8 src/dwb.h
--- a/src/dwb.h	Fri Feb 03 00:23:04 2012 +0100
+++ b/src/dwb.h	Fri Feb 03 11:32:17 2012 +0100
@@ -44,7 +44,7 @@
 #define NO_URL                      "No URL in current context"
 #define NO_HINTS                    "No Hints in current context"
 
-#define HINT_SEARCH_SUBMIT "_dwb_search_submit_"
+#define HINT_SEARCH_SUBMIT "TERMS"
 
 #if GTK_CHECK_VERSION(3, 0, 0)
 #define _HAS_GTK3 1
