diff -ru a/src/view.c b/src/view.c
--- a/src/view.c	2012-04-13 11:20:46.801550379 +0200
+++ b/src/view.c	2012-04-13 11:23:02.466981465 +0200
@@ -875,8 +875,8 @@
     return;
   }
   if (gl == dwb.state.fview) {
-    if ( !(dwb.state.fview = dwb.state.fview->prev) ) {
-      dwb.state.fview = g_list_first(dwb.state.views)->next;
+    if ( !(dwb.state.fview = dwb.state.fview->next) ) {
+      dwb.state.fview = g_list_last(dwb.state.views)->prev;
       if (dwb.state.bar_visible & BAR_VIS_TOP) 
         gtk_widget_show_all(dwb.gui.topbox);
     }
