diff -ru a/src/paps.1 b/src/paps.1
--- a/src/paps.1	2012-07-12 18:24:42.024133884 +0200
+++ b/src/paps.1	2012-07-12 18:25:56.795290161 +0200
@@ -64,6 +64,10 @@
 .B \-\-header\-font=desc
 Set the header font description. Default is Monospace Bold 12.
 .TP
+.B \-\-header\-date\-format=fmt
+Set the header date format. Default is %c (see 
+.BR strftime (3)).
+.TP
 .B \-\-markup
 Interpret the text as pango markup.
 .TP
diff -ru a/src/paps.c b/src/paps.c
--- a/src/paps.c	2012-07-12 18:23:10.712721488 +0200
+++ b/src/paps.c	2012-07-12 18:27:48.543684870 +0200
@@ -36,6 +36,7 @@
 #define DEFAULT_FONT_SIZE	"12"
 #define HEADER_FONT_FAMILY	"Monospace Bold"
 #define HEADER_FONT_SCALE	"12"
+#define HEADER_DATE_FORMAT	"%c"
 #define MAKE_FONT_NAME(f,s)	f " " s
 
 typedef enum {
@@ -88,6 +89,7 @@
   PangoDirection pango_dir;
   gchar *filename;
   gchar *header_font_desc;
+  gchar *header_date_format;
   gdouble lpi;
   gdouble cpi;
 } page_layout_t;
@@ -284,6 +286,7 @@
   int top_margin = 36, bottom_margin = 36, right_margin = 36, left_margin = 36;
   gchar *font = MAKE_FONT_NAME (DEFAULT_FONT_FAMILY, DEFAULT_FONT_SIZE), *encoding = NULL;
   gchar *header_font_desc = MAKE_FONT_NAME (HEADER_FONT_FAMILY, HEADER_FONT_SCALE);
+  gchar *header_date_format = HEADER_DATE_FORMAT;
   page_layout_t page_layout;
   GOptionContext *ctxt = g_option_context_new("[text file]");
   GOptionEntry entries[] = {
@@ -293,6 +296,7 @@
     {"columns", 0, 0, G_OPTION_ARG_INT, &num_columns, "Number of columns output. (Default: 1)", "NUM"},
     {"font", 0, 0, G_OPTION_ARG_STRING, &font, "Set the font description. (Default: Monospace 12)", "DESC"},
     {"header-font", 0, 0, G_OPTION_ARG_STRING, &header_font_desc, "Set the header font description. (Default: Monospace Bold 12)", "DESC"},
+    {"header-date-format", 0, 0, G_OPTION_ARG_STRING, &header_date_format, "Set the header date format. (Default: %%c)", "DESC"},
     {"rtl", 0, 0, G_OPTION_ARG_NONE, &do_rtl, "Do rtl layout.", NULL},
     {"paper", 0, 0, G_OPTION_ARG_CALLBACK, _paps_arg_paper_cb,
      "Choose paper size. Known paper sizes are legal,\n"
@@ -453,6 +457,7 @@
   page_layout.pango_dir = pango_dir;
   page_layout.filename = filename_in;
   page_layout.header_font_desc = header_font_desc;
+  page_layout.header_date_format = header_date_format;
 
   /* calculate x-coordinate scale */
   if (page_layout.cpi > 0.0L)
@@ -1178,7 +1183,7 @@
 
   t = time(NULL);
   tm = *localtime(&t);
-  strftime(date, 255, "%c", &tm);
+  strftime(date, 255, page_layout->header_date_format, &tm);
   header = g_strdup_printf("<span font_desc=\"%s\">%s</span>\n"
                            "<span font_desc=\"%s\">%s</span>\n"
                            "<span font_desc=\"%s\">Page %d</span>",
