diff -ru a/src/paps.1 b/src/paps.1
--- a/src/paps.1	2012-07-12 18:25:56.795290161 +0200
+++ b/src/paps.1	2012-07-12 18:29:14.185009403 +0200
@@ -68,6 +68,9 @@
 Set the header date format. Default is %c (see 
 .BR strftime (3)).
 .TP
+.B \-\-header\-rule\-thickness=val
+Set the thickness of the header separation rule. Default is 0 (i.e. as thin as possible).
+.TP
 .B \-\-markup
 Interpret the text as pango markup.
 .TP
diff -ru a/src/paps.c b/src/paps.c
--- a/src/paps.c	2012-07-12 18:27:48.543684870 +0200
+++ b/src/paps.c	2012-07-12 18:59:23.824011151 +0200
@@ -37,6 +37,7 @@
 #define HEADER_FONT_FAMILY	"Monospace Bold"
 #define HEADER_FONT_SCALE	"12"
 #define HEADER_DATE_FORMAT	"%c"
+#define HEADER_RULE_THICKNESS 0
 #define MAKE_FONT_NAME(f,s)	f " " s
 
 typedef enum {
@@ -90,6 +91,7 @@
   gchar *filename;
   gchar *header_font_desc;
   gchar *header_date_format;
+  double header_rule_thickness;
   gdouble lpi;
   gdouble cpi;
 } page_layout_t;
@@ -287,6 +289,7 @@
   gchar *font = MAKE_FONT_NAME (DEFAULT_FONT_FAMILY, DEFAULT_FONT_SIZE), *encoding = NULL;
   gchar *header_font_desc = MAKE_FONT_NAME (HEADER_FONT_FAMILY, HEADER_FONT_SCALE);
   gchar *header_date_format = HEADER_DATE_FORMAT;
+  double header_rule_thickness = HEADER_RULE_THICKNESS;
   page_layout_t page_layout;
   GOptionContext *ctxt = g_option_context_new("[text file]");
   GOptionEntry entries[] = {
@@ -297,6 +300,7 @@
     {"font", 0, 0, G_OPTION_ARG_STRING, &font, "Set the font description. (Default: Monospace 12)", "DESC"},
     {"header-font", 0, 0, G_OPTION_ARG_STRING, &header_font_desc, "Set the header font description. (Default: Monospace Bold 12)", "DESC"},
     {"header-date-format", 0, 0, G_OPTION_ARG_STRING, &header_date_format, "Set the header date format. (Default: %%c)", "DESC"},
+    {"header-rule-thickness", 0, 0, G_OPTION_ARG_DOUBLE, &header_rule_thickness, "Set the thickness of the header separation rule. (Default: 0 (i.e. as thin as possible))", "REAL"},
     {"rtl", 0, 0, G_OPTION_ARG_NONE, &do_rtl, "Do rtl layout.", NULL},
     {"paper", 0, 0, G_OPTION_ARG_CALLBACK, _paps_arg_paper_cb,
      "Choose paper size. Known paper sizes are legal,\n"
@@ -458,6 +462,7 @@
   page_layout.filename = filename_in;
   page_layout.header_font_desc = header_font_desc;
   page_layout.header_date_format = header_date_format;
+  page_layout.header_rule_thickness = header_rule_thickness;
 
   /* calculate x-coordinate scale */
   if (page_layout.cpi > 0.0L)
@@ -1256,9 +1261,10 @@
   /* header separator */
   line_pos = page_layout->page_height - page_layout->top_margin - page_layout->header_height - page_layout->header_sep / 2;
   g_string_append_printf(ps_pages_string,
-                         "%d %f moveto %d %f lineto 0 setlinewidth stroke\n",
+                         "%d %f moveto %d %f lineto %f setlinewidth stroke\n",
                          page_layout->left_margin, line_pos,
-                         page_layout->page_width - page_layout->right_margin, line_pos);
+                         page_layout->page_width - page_layout->right_margin, line_pos,
+                         page_layout->header_rule_thickness);
 
   return logical_rect.height;
 }
